Documentation/workflow/EventPipeDeveloperGuide.md

```diff
@@ -0,0 +1,105 @@
+
+# Event Pipe Developer Guide
+
+The goal of this guide is to give a good introduction for new contributors who would like to develop Event Pipe.
+
+## What is Event Pipe?
+
+It's new, cross-platform mechanism for fast event tracing in .NET Core. [Here](https://github.com/dotnet/designs/blob/master/accepted/cross-platform-performance-monitoring.md) you can read more about its design. 
+
+The key things are:
+
+* no OS-specific dependencies
+* writer writes to a binary file, a reader reads directly from the file
+* it's not a replacement, but rather an extension
+
+## Where do I find the code?
+
+The code is split into two repositories: [CoreCLR](https://github.com/dotnet/coreclr) and [PerfView](https://github.com/Microsoft/perfview). The core of Event Pipe is implemented in CoreCLR. The most important files are:
+
+CoreCLR (the EventPipes are built-in feature of CoreCLR):
+
+* [src/vm/eventpipe.cpp](../../src/vm/eventpipe.cpp)
+* [src/vm/eventpipebuffer.cpp](../../src/vm/eventpipebuffer.cpp)
+* [src/vm/eventpipebuffermanager.cpp](../../src/vm/eventpipebuffermanager.cpp)
+* [src/vm/eventpipeconfiguration.cpp](../../src/vm/eventpipeconfiguration.cpp)
+* [src/vm/eventpipeevent.cpp](../../src/vm/eventpipeevent.cpp)
+* [src/vm/eventpipeeventinstance.cpp](../../src/vm/eventpipeeventinstance.cpp)
+* [src/vm/eventpipefile.cpp](../../src/vm/eventpipefile.cpp) - creates and writes to Event Pipe file. It also writes the file header and maintains the forward references.
+* [src/vm/eventpipejsonfile.cpp](../../src/vm/eventpipejsonfile.cpp) - allows to use JSON instead of binary file format. Used only for debugging purposes.
+* [src/vm/eventpipeprovider.cpp](../../src/vm/eventpipeprovider.cpp)
+* [src/vm/fastserializer.cpp](../../src/vm/fastserializer.cpp) - fast binary serializer used by `EventPipeFile` to store data
+* [src/scripts/genEventPipe.py](../../src/scripts/genEventPipe.py) - Python script run at build time, generates all event definitions
+* [tests/src/tracing/eventpipetrace/EventPipeTrace.cs](../../tests/src/tracing/eventpipetrace/EventPipeTrace.cs) - Tests. As of today (18th of January 2018) these testes are not run by the CoreCLR CI.
+
+PerfView repository is the home for `Microsoft.Diagnostics.Tracing.TraceEvent` Library, which provide client classes that allow to consume Event Pipe:
```

Comment:
Gramatical: 
* 'which provide' -> 'which provides'
* 'that allow to consume EventPipe' -> 'for consuming EventPipe events'